src/components/diagrams/singleLineDiagram/single-line-diagram.js

```diff
@@ -267,10 +267,20 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
 
     const [loadingState, updateLoadingState] = useState(false);
 
+    const [oneNodeBuilding, setOneNodeBuilding] = useState(false);
+
+    const treeModel = useSelector(
+        (state) => state.networkModificationTreeModel
+    );
+
     const MenuLine = withLineMenu(BaseEquipmentMenu);
 
     const theme = useTheme();
 
+    useEffect(() => {
+        setOneNodeBuilding(treeModel.isOneNodeBuilding);
+    }, [treeModel]);
+
```

Comment:
Maybe use a custom hook for oneNodeBuilding (with the useState, useSelector and useEffect) ?
because same code is repeated in multiple components (create-node-menu, network-modification-node-editor, line-menu, network-table ...)

Reply:
Yes that's cleaner. Done